libnymea-core/hardware/network/networkdevicediscoveryimpl.cpp

```diff
@@ -352,56 +355,83 @@ void NetworkDeviceDiscoveryImpl::pingAllNetworkDevices()
 
         qCDebug(dcNetworkDeviceDiscovery()) << "Verifying network interface" << networkInterface.name() << networkInterface.hardwareAddress() << "...";
         foreach (const QNetworkAddressEntry &entry, networkInterface.addressEntries()) {
-            qCDebug(dcNetworkDeviceDiscovery()) << "  Checking entry" << entry.ip().toString();
 
             // Only IPv4
             if (entry.ip().protocol() != QAbstractSocket::IPv4Protocol)
                 continue;
 
+            qCDebug(dcNetworkDeviceDiscovery()) << "  Checking entry" << entry.ip().toString();
             qCDebug(dcNetworkDeviceDiscovery()) << "    Host address:" << entry.ip().toString();
             qCDebug(dcNetworkDeviceDiscovery()) << "    Broadcast address:" << entry.broadcast().toString();
             qCDebug(dcNetworkDeviceDiscovery()) << "    Netmask:" << entry.netmask().toString();
-            quint32 addressRangeStart = entry.ip().toIPv4Address() & entry.netmask().toIPv4Address();
-            quint32 addressRangeStop = entry.broadcast().toIPv4Address() | addressRangeStart;
-            quint32 range = addressRangeStop - addressRangeStart;
+
+            // Store the own address of this network interface in any case,
+            // since we don't want to ping our self
+            ownAddresses.append(entry.ip());
+
+            TargetNetworkRange networkRange;
+            networkRange.networkInterface = networkInterface;
+            networkRange.addressRangeStart = entry.ip().toIPv4Address() & entry.netmask().toIPv4Address();
+            networkRange.addressRangeStop = entry.broadcast().toIPv4Address() | networkRange.addressRangeStart;
+            networkRange.range = networkRange.addressRangeStop - networkRange.addressRangeStart;
 
             // Let's scan only 255.255.255.0 networks for now
-            if (range > 255)
+            if (networkRange.range > 255)
```

Comment:
could just be the `if (entry.prefixLength() < 24)` instead of calculating the range